spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -62,7 +62,7 @@ public ConsumerDestination provisionConsumerDestination(String name, String grou
 		// Generate anonymous random group, if one not provided
 		String subscription = !StringUtils.hasText(group) ?
 				"anonymous." + name + "." + UUID.randomUUID().toString()
-				: group;
+				: name + "." + group;
```

Comment:
I think this is not good. I don't believe that you can make end-users to always create subscriptions with this pattern.
We only have to say that `consumerGroup` has to be supplied with the subscription name.
More over this your pattern is going to be included in just general `group` solution.
Otherwise you can't support any arbitrary names and your provisioner is going to create extra subscription when end-user would like to rely on existing one.

Does it make sense to you?

Reply:
I can see that point of view too, but `AbstractBinderTests.testSendAndReceiveMultipleTopics` will fail without this because it uses the same consumer group name with multiple topics.

Reply:
Huh? So, with this nasty hack you are going to meet wrong test requirements? 😕 

Why can't you specify consumer group properties there with the desired `[TOPIC].[SUBSCRIPTION]` pattern?

Reply:
Haha, it's not my test - `org.springframework.cloud.stream.binder.AbstractBinderTests.testSendAndReceiveMultipleTopics`. I looked at the test as the binder contract definition. The test implies that consumer group names are scoped to a destination, which we can only support by adding the prefix to the subscription name. If that's in fact, not the contract, the test in `spring-cloud-stream-binder-test` should be fixed. I can ignore that particular test for now.

Reply:
OK sounds good.
In Pub/Sub subscription names must be unique to a project. I'm not sure why the names aren't scoped to a topic, but that's just how it is right now. @kir-titievsky?

Reply:
Hm. OK. So, this is a specific of Pub/Sub and we need take it into account, but still not fix the core because end-users aware of such a global perspective and just fix the test in really scope of this project and not SCSt core.